src/io/flutter/inspector/Location.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2018 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.inspector;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.intellij.openapi.vfs.LocalFileSystem;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.xdebugger.XSourcePosition;
+import com.intellij.xdebugger.impl.XSourcePositionImpl;
+import io.flutter.utils.JsonUtils;
+
+import java.util.ArrayList;
+
+public class Location {
+  final JsonObject json;
+  Location parent;
+
+  public Location(JsonObject json, Location parent) {
```

Comment:
we're not using the `parent` param here?

Reply:
the usage isn't obvious but we are on line 29. the behavior is the file from the parent is used if a fileName is not specified as is the case for parameter locations.
Notice that on line 71 we pass in parent as "this".